### PR TITLE
Fabric upgrade

### PR DESCRIPTION
**Pull request by jagpreetsinghsasan**
Tested on fabric base image version  at 1.4.4 and zookeeper, kafka & couchDB version at 0.4.18
#47 

Signed-off-by: Sasan <jagpreet.singh.sasan@accenture.com>

### CODE REVIEW

File: docs/source/faq.md [hunk 1]
```diff
@@ -43,7 +43,7 @@ Unfortunately, there are no existing training for using the Blockchain Automatio
 Yes, you can add additional nodes to a running DLT network using the Blockchain Automation Framework (BAF). Unfortunately, BAF does not support removing nodes in a running DLT network, but this significant feature is in our future roadmap, and we will add this feature in a future release.
 
 ### Does the Blockchain Automation Framework support multiple versions of Fabric and Corda? What are the minimum versions for Fabric and Corda supported in the Blockchain Automation Framework?
-No, unfortunatley, the Blockchain Automation Framework currently only supports version 1.4.0 for Fabric and version 4.0 for Corda as minimum versions, and will only support future higher versions for Fabric and Corda.
+No, unfortunatley, the Blockchain Automation Framework currently only supports version 1.4.0 & 1.4.4 for Fabric and version 4.0 for Corda as minimum versions, and will only support future higher versions for Fabric and Corda.
```
Comment (sownak) [inline docs/source/faq.md]: Change Corda to 4.1

File: docs/source/operations/upgrade_fabricnetwork.md [hunk 2]
```diff
@@ -0,0 +1,26 @@
+<a name = "upgrading-fabric"></a>
+# Upgrading Hyperledger Fabric version
+
+- [Pre-requisites](#pre_req)
+- [Modifying image versions](#modify_image_version)
+
+<a name = "pre_req"></a>
+## Pre-requisites
+Hyperledger Fabric image versions, which are compatible with the target fabric version need to be known. Here's the list of the images for which the versions should be known:  
+kafka image-- hyperledger/fabric-kafka:`targer_version`
```
Comment (sownak) [inline docs/source/operations/upgrade_fabricnetwork.md]: change this to a table, saying "For example, for Fabric v1.4.4, these are the image tags of the supporting docker images"

File: docs/source/operations/upgrade_fabricnetwork.md [hunk 3]
```diff
@@ -0,0 +1,26 @@
...
+
+<a name = "pre_req"></a>
+## Pre-requisites
+Hyperledger Fabric image versions, which are compatible with the target fabric version need to be known. Here's the list of the images for which the versions should be known:  
+kafka image-- hyperledger/fabric-kafka:`targer_version`  
+zookeeper image-- hyperledger/fabric-zookeeper:`target_version`  
+couchDB image-- hyperledger/fabric-couchdb:`target_version`  
+
+---
+**NOTE:** The upgrade based solely on the base image version change will work only if the fabric upgrade requires the change of image versions only
```
Comment (sownak) [inline docs/source/operations/upgrade_fabricnetwork.md]: "This change only upgrades the docker images, any other configuration changes is not covered by this guide. Please refer to Fabric documentation for any specific configuration changes."

File: docs/source/operations/upgrade_fabricnetwork.md [hunk 4]
```diff
@@ -0,0 +1,26 @@
...
+
+---
+
+<a name = "modify_image_version"></a>
+## Modifying image versions
+The network.yaml [here](https://github.com/hyperledger-labs/blockchain-automation-framework/tree/master/platforms/hyperledger-fabric/configuration/samples/network-fabricv2.yaml) should be updated with the required version tag under `network.version` for upgrading the base images of CA, orderer and peer.
+
```
Comment (sownak) [inline docs/source/operations/upgrade_fabricnetwork.md]: Add an example like we did in networkyaml readthedocs
```
network:
  version:
```

File: docs/source/operations/upgrade_fabricnetwork.md [hunk 5]
```diff
@@ -0,0 +1,26 @@
...
+2 files need to be edited in order to support version change for kafka, zookeeper and couchDB
```
Comment (sownak) [inline docs/source/operations/upgrade_fabricnetwork.md]: Create a table instead,
with file name, and the tag to change. This is not very clear.

File: platforms/hyperledger-fabric/configuration/roles/create/anchorpeer/tasks/nested_anchorpeer.yaml [hunk 6]
```diff
@@ -32,6 +32,8 @@
     git_branch: "{{ org.gitops.branch }}"
     charts_dir: "{{ org.gitops.chart_source }}"
     vault: "{{ org.vault }}"
+    fabrictools_image: "hyperledger/fabric-tools:{{ fabric_version }}"
```
Comment (sownak) [inline platforms/hyperledger-fabric/configuration/roles/create/anchorpeer/tasks/nested_anchorpeer.yaml]: directly use network.version  and network.docker.url here, just to reduce the number of file changes.

File: platforms/hyperledger-fabric/configuration/roles/create/peers/tasks/main.yaml [hunk 7]
```diff
@@ -21,6 +21,9 @@
     name: "{{ item.name | lower }}"
     type: "value_peer"
     component_name: values-{{ peer.name }}
+    couchdb_image: "hyperledger/fabric-couchdb:{{ couchdb_image_version[fabric_version] }}"
+    peer_image: "hyperledger/fabric-peer:{{ fabric_version }}"
```
Comment (sownak) [inline platforms/hyperledger-fabric/configuration/roles/create/peers/tasks/main.yaml]: same here

Comment (jagpreetsinghsasan) [reply platforms/hyperledger-fabric/configuration/roles/create/peers/tasks/main.yaml]: I have used fabric_version as we dont have variable nesting support in jinja like {{ var1{{ var2 }} }}
Hence I am fetching it as a variable from a level above (deploy-network.yaml)

**Pull request merged**